modulemd/v2/modulemd-component.c

```diff
@@ -48,6 +48,14 @@ enum
 static GParamSpec *properties[N_PROPS];
 
 
+gboolean
+modulemd_component_equals_wrapper (const void *a, const void *b)
+{
+  return modulemd_component_equals ((ModulemdComponent *)a,
```

Comment:
We want to `assert()` on programming errors here.

Also, GLib has a cast mechanism for objects that we should prefer:

```suggestion
  g_return_val_if_false (MODULEMD_IS_COMPONENT(a), FALSE);
  g_return_val_if_false (MODULEMD_IS_COMPONENT(b), FALSE);
  
  return modulemd_component_equals (MODULEMD_COMPONENT(a),
```